packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -631,3 +631,85 @@ export class KeybindingWidget extends ReactWidget {
     }
 
 }
+/**
+ * Dialog used to edit keybindings, and reset custom keybindings.
+ */
+class EditKeybindingDialog extends SingleTextInputDialog {
+
+    /**
+     * The keybinding item in question.
+     */
+    protected item: KeybindingItem;
+
+    /**
+     * HTMLButtonElement used to reset custom keybindings.
+     * Custom keybindings have a `User` scope (exist in `keymaps.json`).
+     */
+    protected resetButton: HTMLButtonElement | undefined;
+
+    constructor(
+        @inject(SingleTextInputDialogProps) protected readonly props: SingleTextInputDialogProps,
+        @inject(KeymapsService) protected readonly keymapsService: KeymapsService,
+        item: KeybindingItem
+    ) {
+        super(props);
+        this.item = item;
+        // Add the `Reset` button if the command currently has a custom keybinding.
+        if (this.item.source &&
+            this.getRaw(this.item.source) === KeybindingScope[1].toLocaleLowerCase()) {
+            this.appendResetButton();
+        }
+    }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        if (this.resetButton) {
+            this.addResetAction(this.resetButton, 'click');
+        }
+    }
+
+    /**
+     * Add `Reset` action used to reset a custom keybinding, and close the dialog.
+     * @param element {HTMLElement} the HTML element in question.
+     * @param additionalEventTypes {K[]} additional event types.
+     */
+    protected addResetAction<K extends keyof HTMLElementEventMap>(element: HTMLElement, ...additionalEventTypes: K[]): void {
+        this.addKeyListener(element, Key.ENTER, () => {
+            this.reset();
+            this.close();
+        }, ...additionalEventTypes);
+    }
+
+    /**
+     * Create the `Reset` button, and append it to the dialog.
+     * @returns the `Reset` button.
+     */
+    protected appendResetButton(): HTMLButtonElement {
+        // Create the `Reset` button.
+        this.resetButton = this.createButton('Reset');
+        // Add the `Reset` button to the dialog control panel, before the `Accept` button.
+        this.controlPanel.insertBefore(this.resetButton, this.acceptButton!);
+        this.resetButton.title = 'Reset Keybinding';
+        this.resetButton.classList.add('secondary');
+        return this.resetButton;
+    }
+
+    /**
+     * Perform keybinding reset.
+     */
+    protected reset(): void {
+        // Extract the raw id from the keybinding item (without fuzzy matching).
+        const id = this.getRaw(this.item.id);
+        // Remove the custom keybinding, resetting it to its default value.
+        this.keymapsService.removeKeybinding(id);
+    }
+
+    /**
+     * Extract the raw value from a string (without fuzzy matching).
+     * @param a the string value.
+     */
+    protected getRaw(a: string): string {
```

Comment:
Minor: Some methods returning something, you put it in the header section, but here, you did not put the return in the header comment ! @return ....